storage/client.go

```diff
@@ -308,6 +308,15 @@ func (c Client) buildCanonicalizedResourceTable(uri string) (string, error) {
 		cr += u.EscapedPath()
 	}
 
+	params, err := url.ParseQuery(u.RawQuery)
+
+	// search for "comp" parameter, if exists then add it to canonicalizedresource
+	for key := range params {
```

Comment:
I'm unclear what you mean by saying use Add? For which part?

As for the "comp" part, that is required when generating the hmac. Other parameters (if any exist) cannot be included, but "comp" is required.  Was a long hard struggle figuring that out.